### ISSUE TITLE
pyside2 setBacground on an item breaks model test

### ISSUE THREAD

**Issue by p0las** [pytest-dev/pytest-qt#189]
in pyside2 setBackground on standartItem requires a brush. However it seems that qtmodeltester fixture assumes it to be QColor

```python
model = QtGui.QStandardItemModel()
item = QtGui.QStandardItem("test")
model.appendRow([item])
brush = QtGui.QBrush(QtGui.QColor("#DDDDDD"), QtCore.Qt.SolidPattern)
item.setBackground(brush)
qtmodeltester.check(model)
```

```
E           AssertionError: PySide.QtCore.Qt.ItemDataRole.BackgroundColorRole
E           assert (<PySide.QtGui.QBrush(QColor(AHSV 1, -1, 0, 0.235294) , NoBrush )   at 0x000000000A0C3608> == None or False)
E            +  where False = isinstance(<PySide.QtGui.QBrush(QColor(AHSV 1, -1, 0, 0.235294) , SolidPattern )   at 0x000000000A0C3608>, <type 'PySide.QtGui.QColor'>)
```

EDIT by @nicoddemus: formatting

**Comment by nicoddemus** [comment]
Hi @p0las,

The `qtmodeltester` fixture was a direct translation from C++ of the standard [Model Test](http://wiki.qt.io/Model_Test) so I don't know if it is `pytest-qt`'s fault.

I see that the C++ signature of [setBackground/background](http://doc.qt.io/qt-5/qstandarditem.html#setBackground) indeed receives/returns a `QBrush`.

One solution might be to accept both `QColor` and `QBrush` for that data role:

https://github.com/pytest-dev/pytest-qt/blob/5d47c68d12cca561d4739d811aff60746b13ca19/pytestqt/modeltest.py#L445-L453

So changing the relevant line to:

```python
(qt_api.QtCore.Qt.BackgroundColorRole, (qt_api.QtGui.QColor, qt_api.QtGui.QBrush)), 
```

Might be all we need. Would you like to contribute a PR with that?

**Comment by The-Compiler** [comment]
I don't think the Qt C++ modeltest has seen much love since Qt 4 days - I think `QBrush` is in fact the only valid value for those roles, at least on Qt 5.

Also, `BackgroundColorRole` and `TextColorRole` are obsolete according to [the docs]( http://doc.qt.io/qt-5/qt.html#ItemDataRole-enum) and should be `BackgroundRole` and `ForegroundRole` instead.

**Comment by nicoddemus** [comment]
I see thanks @The-Compiler.

I'm OK with my original suggestion of accepting both to make maintenance easy; the fact that PyQt5 accepts `QColor` as `BackgroundRole` hints that this is OK; the fact that `QBrush` has a non-explicit constructor which accepts `QColor` indicates that this valid in C++ as well.

**Comment by p0las** [comment]
@nicoddemus to be honest I just started to use pytest-qt and didn't know what to expect. I'm happy to try to contribute but as I just started to use it I don't want to break it :-). I'm also switching from unittest to pytest (to be able to use the fixtures) so there is a lot I'm not fully familiar with.

**Comment by nicoddemus** [comment]
> as I just started to use it I don't want to break it 

Fear not, we have tons of tests in place to prevent just that. 😉 Feel free to contribute whenever you feel comfortable. 👍

**Comment by nicoddemus** [comment]
Fixed by #190
